clang/lib/Sema/SemaChecking.cpp

```diff
@@ -3587,6 +3587,30 @@ Sema::CheckBuiltinFunctionCall(FunctionDecl *FDecl, unsigned BuiltinID,
     }
     break;
   }
+
+  case Builtin::BI__builtin_allow_sanitize_check: {
+    Expr *Arg = TheCall->getArg(0);
+    // Check if the argument is a string literal.
+    const StringLiteral *SanitizerName =
+        dyn_cast<StringLiteral>(Arg->IgnoreParenImpCasts());
```

Comment:
During the review of `__builtin_allow_runtime_check()` it was suggested that ``__builtin_allow_runtime_check` can use various conditions for lowering.
Maybe we can reserve sanitizer names and lower that one as needed?